students/jeff_shabani/session07/html_render.py

```diff
@@ -8,21 +8,205 @@
 # This is the framework for the base class
 class Element(object):
 
-    def __init__(self, content=None):
-        self.content = str(content)
+    indent = " "*2
+
+    def __init__(self, content=None, tag=None, **attrs):
+        self.attrs = attrs
+        if tag:
+            self.tag = tag
+        else:
+            self.tag = ''
+        if content:
+            self.content = content
+        else:
+            self.content = ''
 
     def append(self, new_content):
```

Comment:
This one should also be a lot simpler, you're only adding additional lines of content to the content list, so something like this:

```
    def append(self, new_content):
        self.content.append(new_content)
```